core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -191,7 +191,10 @@ public RelFieldCollation toFieldCollation(Map<String, Object> map) {
   }
 
   public RelDistribution toDistribution(Object o) {
-    return RelDistributions.ANY; // TODO:
+    final Map map = (Map) o;
```

Comment:
Is there a need to check if Object is a Map?

Reply:
The style of RelJson (which I agree with) is to assume that objects are of the correct type, i.e. they match that produced by the corresponding toJson method (in this case `toJson(RelDistribution)`).

I'd add javadoc for the `toDistribution` method describing the contract.

Reply:
+1. That would be much clear if have a java doc to link to the corresponding `toJason` method, in which a map Object is returned.

Reply:
Sure, that's better